examples/matmul/run.cpp

```diff
@@ -726,23 +778,30 @@ Kernel selectMatmul(Context &ctx, int version,
     LOG(kDefLog, kInfo, "nWorkgroups: ( %s )", toString(nWorkgroups).c_str());
     KernelCode matmul = createMatmulWithTranspose(kShaderMatmulWithTranspose, M, K, N, BM, BK, BN, TM, TN,
 						  /*wgSize*/ wgSize,
-						  kf32);
+						  kf);
     kernel = createKernel(ctx, matmul, bindings,
                           /*nWorkgroups*/ nWorkgroups);
   }
   return kernel;
 }
 
+template<class precision=float>
```

Comment:
After learning my lesson from type-templated early iterations of gemma.cpp my bias here is to implement the core implementations with precision types at the value level as a default unless there's a rationale otherwise, but only for if there's a strong case.

Templated overloads are okay as a more typesafe convenience API sugar - it's easy to wrap a value-level core implementation in a more strongly templated wrapper. It's a bigger refactoring to go in the other direction and unwind a core implementation into value level.

My suggestion would be, unless there's some value I'm missing.
- Either have `runTest` take a precision argument *or* have a separate `runTest` (e.g. `runtest16`) for alternative precision values.
- avoid having `getPrecision` to pull out values from types, remove them entirely unless there's something we need them for.
-